gamemodes/horde/entities/entities/npc_satchel_horde/init.lua

```diff
@@ -117,7 +117,4 @@ function ENT:Use(ply)
     self:EmitSound(defuse, 60)
     self:Remove()
     ply:GiveAmmo(1, "slam", true)
-    if not IsValid(ply:GetWeapon(self.SWEP)) then
```

Comment:
Can you change this to only allow pickup if you have self.SWEP equipped? Can be checked with https://wiki.facepunch.com/gmod/Player:HasWeapon

Same for the tripmine